lib/manager/cocoapods/readme.md

```diff
@@ -0,0 +1,232 @@
+## Overview
+
+#### Name of package manager
+
+Cocoapods
+
+---
+
+#### What language does this support?
+
+Swift, Objective-C
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+Carthage, Swift Package Manager
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+Podfile
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+`['Podfile']`\*
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+No
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+Unlikely
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+No
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+Package file uses ruby syntax.
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+Using regex
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+It distinguishes dependencies for different targets (testing target, target for TV OS etc)
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+Dependency indicated by keyword `pod` and name with version.
+
+Also, it's possible to use a local podspec file on the machine, or in the root of library repo.
+In case of the repo you also can specify branch, tag or commit.
+
+```
+pod 'AFNetworking', '~> 2.6'
+pod 'Alamofire', :path => '~/Documents/Alamofire'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :branch => 'dev'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :tag => '3.1.1'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :commit => '0f506b1c45'
+```
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+Cocoapods supports dependencies from git source and local path.
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+Package file use semver 2.0. ([details](https://semver.org))
```

Comment:
Even though it's using semver 2.0, it looks like you can define "short" versions? e.g. https://github.com/CocoaPods/Specs/tree/master/Specs/5/5/e/Watchdog has versions like `5.1` instead of `5.1.0`

Reply:
Yes, it does. People set the version, so I think we need to handle this case too.